### ISSUE TITLE
System to generate construction sequencing animations

### ISSUE THREAD

**Issue by Moult** [opencax/GSoC#42]
## Outline

Often, construction firms need to visualise animations of construction sequencing. A project timeline will be created, and related to individual model elements. For example, when a concrete slab is poured, it is linked to a 3D object called a slab. We need the ability to automatically generate animations from Blender where objects appear / disappear in various different ways when they start / end their task in the project timeline.

## Expected Outcome

A series of small scripts that take objects and can automatically animate the visibility, locations, or staggered appearances of building elements, as well as sub elements, and basic scripts that correlate real world time to animation frames, and frames per second, and generate an animated timeline bar in various styles.

## Future Possibilities 

This animation system can be then used from BIM models either in Blender, FreeCAD, or via other software altogether, so it has quite a large impact on the ecosystem.

### Skills

 - Basic knowledge of the principles of animation (keyframing)
 - Basic Blender animation (you can do some tutorials and get up to speed pretty quick)
 - Python
 - Artistic sense! We should offer beautiful and elegant animations!

### Difficulty 

Easy

### Additional Information

* Potential mentor(s): Dion Moult
* Organization website: https://ifcopenshell.org
* Communication channels: https://community.osarch.org , ##architect on Freenode , and https://github.com/IfcOpenShell/IfcOpenShell/issues

**Comment by Szymon-Budziak** [comment]
Hello, I am interested in working on this project for GSoC. Is is still available?

**Comment by Moult** [comment]
@Szymon-Budziak yes, this is still available, but you've got to be quick as the deadline is 14th of April. Catch me online on IRC via the OSArch.org live chat (##architect on Freenode) and let's discuss! Also, get a head-start and start writing your proposal!

**Comment by samadpls** [comment]
I am interested in working on this project as it aligns with my experience. May I inquire about the best way to reach you for further discussion?

**Comment by Moult** [comment]
Absolutely! Feel free to reach out on osarch.org/chat and/or dion@thinkmoult.com and let's have a chat further!

**Comment by jaymeandretta** [comment]
Nice feature! I'd like to participate! I've never applied for GSoC. Is there anything I should do before applications start? @Moult

**Comment by Moult** [comment]
@jaymeandretta indeed! Reach out on osarch.org/chat - introduce yourself, start looking around the codebase and familiarise yourself with the code!

**Comment by niuniuzhao** [comment]
Hello everyone, Hi @Moult ! I am an architecture student and a GSoC2024 applicant. I'm very interested in this project because it seems like a great fit! Could you please advise whether the scope of the project would be within the reach for someone with sophomore computer science knowledge - wide introduction + data structures (and a lot of motivation, of course)? I love a challenge but also don't want to overshoot too much. :) Thanks!

**Comment by Moult** [comment]
G'day @niuniuzhao ! Yes indeed, this is a pretty straightforward GSoC project! It would be great to start off with a 1-page summary of any thoughts you had about how to tackle this and what you'd like to build, and when you're ready let's meet up for a voice call!

**Comment by niuniuzhao** [comment]
Hi @Moult , I hope you are feeling better! I have done up a 1 page summary, and it would be great if we could do a quick call, maybe between 10am-1pm Sydney time? Thanks!

**Comment by Moult** [comment]
@niuniuzhao absolutely! I'm free today if you'd like, just join https://osarch.org/chat and let's meet up!

**Comment by creativechirag26** [comment]
Hey @Moult , 

I am Chirag and I went through the project idea, I know its a GSOC Project, but if you don't mind, I would like to gain some knowledge by contributing into this project.

**Comment by Moult** [comment]
Dion Moult

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by DraKen0009** [comment]
Hello @Moult  , I would like to contribute to issue, since it was the only easy one with python requirement. I haven't really worked with animation specifically but I'm decently good with python so I can learn few new things with it working on it. 
Also can you please guide me to any contributor-guide available for the org and where can I find issues related to python. 

Thanks for time.

**Comment by Moult** [comment]
G'day! https://docs.bonsaibim.org/ and https://github.com/IfcOpenShell/IfcOpenShell/issues should help!

**Comment by basvi-chunara** [comment]
Hello @Moult , I would like to know if this project is still available to work on? 
I'm really looking out for projects like these to increase my open-source knowledge. Any advice for a beginner ? Should I directly start with project or work on some good-first-issues ? 
Thank you for your time!

**Comment by Moult** [comment]
@basvi-chunara g'day, yes, this is still available.

**Comment by leonstrr** [comment]
Hey @Moult! 
I am interested in this issue, too. Is it generally foreseen that more than 1 contributors work on (this) issue(s)?
Looking forward hearing from you!

**Comment by Moult** [comment]
@leonstrr in general there is only one contributor per project. The only time you have 2 contributors is if the project can be divided into two with clear scope for each :)

Would you be interested in applying?

**Comment by basvi-chunara** [comment]
Hello @Moult, thank you for your response!
I'm interested to work on this project and planning to apply to gsoc 2025 to gain hands-on experience:)
Also, I am open to collaborate (if this project can be divided with two scopes!) 

Would like to know if you've any tips for a strong gsoc application? 
Looking forward to your insights:)

**Comment by Moult** [comment]
Thats great! The strongest application is where the applicant has already actively reached out on our chatroom https://osarch.org/chat and has already installed and tried our software so they know the context of what they're building, and has already written a little bit of code related to the project in some way so that they aren't going in blind, they know what's involved.

More tips here https://opencax.github.io/

Dion Moult

Sent from Proton Mail Android

-------- Original Message --------

**Comment by basvi-chunara** [comment]
Got it! Thanks a lot @Moult

**Comment by AmandeepMandal1077** [comment]
Hi, i would like to know more about this issue.
From what I understand, the task is to develop a set of basic animation scripts that can be applied to models. Essentially, it's pre-built animations—such as bounce, fade in/out, appear, and disappear—that can be easily added to any basic shape model.
am i right.
I would appreciate any additional details or guidance.

**Comment by Moult** [comment]
G'day! The best thing to do is to reach out in the osarch.org/chat and
say hello. Let us know your experience in Blender, or animations, and
Python :)

**Comment by eng-ehab-n** [comment]
Hi
I’m interested in this project for GSoC 2026
Is this project still available?
do you think this is a good for a beginner in python?

**Comment by Arijit-Pal77** [comment]
Hi everyone, I’m new to open-source and to BRL‑CAD, and I’d really like to learn how to start contributing. I’ve been reading about the GSoC continuation projects and exploring the wiki, but I’m still figuring out where to begin. Could you guide me on the first steps like how to pick a project, set up the codebase, or make an initial commit? I’m eager to learn and contribute, even if I start small.
